### PR TITLE
Orders Implementation

### PR DESCRIPTION
**Pull request by tamurtaj85**
Implemented authentication on routes, joi validation, order placement

### CODE REVIEW

File: src/routes/productRoutes.js [hunk 1]
```diff
@@ -1,20 +1,21 @@
 import controller from "../controllers/index.js";
 import validationSchemas from "../validations/index.js";
-import { validator } from "../middleware/validatorMW.js";
-
-const options = { warnings: true };
+import middleware from "../middleware/index.js";
 
 export function routes_Products(app) {
+  app.use("/product?s", middleware.authenticateUser);
```
Comment (bilalk1) [inline src/routes/productRoutes.js:6]: Why we are using `?`

**Comment by bilalk1** [comment]
Please attach the video

File: src/routes/productRoutes.js (hunk 1 above)
Comment (tamurtaj85) [reply src/routes/productRoutes.js:6]: to match either product or products endpoint.

**Comment by tamurtaj85** [comment]
Here is the video recording of all the API end points testing: https://www.loom.com/share/e048da2e4595447a900e43dd7a2a1b72

**Pull request merged**